### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by hooh** [fljot/Gestouch#15]
While using starling gestouch ,  the ram of flash.display.DisplayObject  can not be recycled after be clicked.

I found the issue is caused by  "Gestouch.inputAdapter ||= new NativeInputAdapter(....);"

**Comment by fljot** [comment]
Any more specific steps to reproduce?

**Comment by hooh** [comment]
1. config starling gestouch.
   
   Gestouch.addDisplayListAdapter(starling.display.DisplayObject, new StarlingDisplayListAdapter());
   
   ```
         Gestouch.inputAdapter ||= new NativeInputAdapter(stage);
   
         starlingTouchHitTester = new StarlingTouchHitTester(_starling);
         Gestouch.addTouchHitTester(starlingTouchHitTester, -1);
   ```
2. add a flash.display.Sprite(or other mc, sprite from swc) on the stage, then add mouse click listener to the sprite.
3. click the sprite added just now (the click action is key point),then remove it from the stage.
4. check the profile in flash builder, the sprite can not be retrieved.

Starling Sprite is fine.

**Comment by fljot** [comment]
Good one! Sorry for troubles and thank you for report. Fixed in v0.4.2

Tested with:

```
package
{
    import org.gestouch.core.Gestouch;
    import org.gestouch.input.NativeInputAdapter;

    import flash.display.Graphics;
    import flash.display.Sprite;
    import flash.events.Event;
    import flash.events.MouseEvent;
    import flash.net.LocalConnection;
    import flash.utils.Dictionary;


    public class GestouchMemoryLeakTest extends Sprite
    {
        private var d:Dictionary = new Dictionary(true);
        private var s:Sprite;


        public function GestouchMemoryLeakTest()
        {
            Gestouch.inputAdapter ||= new NativeInputAdapter(stage);

            s = new Sprite();
            const g:Graphics = s.graphics;
            g.beginFill(0xFF0000, 1);
            g.drawRect(0, 0, 100, 100);
            g.endFill();
            s.addEventListener(MouseEvent.CLICK, onClick);
            addChild(s);

            d[s] = true;
            addEventListener(Event.ENTER_FRAME, onEnterFrame);
        }


        private function onEnterFrame(event:Event):void
        {
            forceGC();

            for (var key:* in d)
            {
                trace("sprite still exists...");
                return;
            }

            trace("sprite was GC'ed!");
            removeEventListener(Event.ENTER_FRAME, onEnterFrame);
        }


        private function onClick(event:MouseEvent):void
        {
            s.removeEventListener(MouseEvent.CLICK, onClick);
            removeChild(s);
            s = null;

            forceGC();
        }


        private function forceGC():void
        {
            try
            {
                new LocalConnection().connect('foo');
                new LocalConnection().connect('foo');
            }
            catch (e:*) {}
        }
    }
}
```

Wonder how come Starling sprite was ok though...
